src/core/border_router/routing_manager.hpp

```diff
@@ -500,6 +502,15 @@ class RoutingManager : public InstanceLocator
 
         struct Router
         {
+            // The timeout (in msec) for router staying in active state
+            // before starting the Neighbor Solicitation (NS) probes.
+            static constexpr uint32_t kActiveTimout = OPENTHREAD_CONFIG_BORDER_ROUTING_ROUTER_ACTIVE_CHECK_TIMEOUT;
+
+            static constexpr uint8_t  kMaxNsProbes          = 3;    // Max number of NS probe attempts.
```

Comment:
I think the current implementation implies that `kMaxNsProbes < 255`. Add a static assert for that?

It's unlikely that kMaxNsProbes be set to 255, but just in case.

Reply:
We define `kMaxNsProbes` constant as `uint8_t` so I think we will get a compile-time error if we try to set it to value larger than `255`.